mlrun/model_monitoring/stream_processing_fs.py

```diff
@@ -719,6 +726,46 @@ def do(self, full_event):
         storey_event = storey.Event(body=events, key=endpoint_id)
         return storey_event
 
+    def _update_request_timestamps(
+        self, endpoint_id: str, timestamp: str
+    ) -> datetime.datetime:
+        """Update model endpoint first and last request timestamps. In addition, this function validates that the
+         request time of the current event is later than the previous request time that has already been processed.
+
+        :param endpoint_id: The unique id of the model endpoint.
+        :param timestamp:   Event request time as a string.
+
+        :return: Event request time as a `datetime.datetime` object
+
+        :raise MLRunPreconditionFailedError: If the request time of the current is later than the previous request time.
+        """
+
+        # Convert timestamp into a `datetime.datetime` object
+        timestamp_as_datetime = datetime.datetime.strptime(
+            timestamp[:-6], "%Y-%m-%d %H:%M:%S.%f"
+        )
+
+        if endpoint_id not in self.first_request:
+            # Set time for the first request of the current endpoint
+            self.first_request[endpoint_id] = timestamp
+        if endpoint_id in self.last_request:
+            # Validate that the current request time is later than the previous request
+            last_request_as_timestamp = datetime.datetime.strptime(
+                self.last_request[endpoint_id][:-6], "%Y-%m-%d %H:%M:%S.%f"
+            )
+            if last_request_as_timestamp > timestamp_as_datetime:
```

Comment:
why do you need to convert to timestamp in order of doing the comparison? won't it work in it's native form?

Reply:
I'll change it to the native form. 

Usually, the best practice is to convert the timestamps strings into `datetime` objects and then compare between these objects. Python compare strings in alphabetical order and in that case the older timestamp strings will be "lower" than the newer timestamp strings and we should get a correct result. However, there are cases in which this approach might be tricky and provide invalid results, e.g. 
```
>>> a ='2009-33-10'
>>> b ='2009-11-1'
>>> a>b
True
```
(for more information - https://stackoverflow.com/questions/31350373/comparing-date-strings-in-python/31350422#31350422)

**However, in our case, as you suggested, we can use the native form** because we are always expecting a similar timestamp string structure (e.g. `2023-02-23 08:19:08.919805+00:00`) that should be fine with the native form.